### ISSUE TITLE
ftw quantitative: -r,--rule does not report false positives above PL-1

### ISSUE THREAD

**Issue by EsadCetiner** [coreruleset/go-ftw#635]
When testing individual rules for false positives using the quantitative tests, if the rule is above PL-1 then no false positives are reported:
```
$ ftw quantitative -r 942200 --debug --max-concurrency 100
5:52PM INF ⏳Running quantitative tests with 100 goroutines
5:52PM DBG Using paranoia level: 1
5:52PM DBG Using default corpus cache directory: /home/esadc/.ftw
5:52PM DBG Preparing download of corpus file from https://downloads.wortschatz-leipzig.de/corpora/eng_news_2023_100K.tar.gz
5:52PM DBG filename eng_news_2023_100K-sentences.txt already exists
5:52PM DBG Printing Stats summary
Run 100000 payloads (0 skipped) in 16.90233525s
No false positives detected with the passed corpus
```

However if I test against a specific paranoia level, I can clearly see false positives being reported:
```
$ ftw quantitative -P 2 --debug --max-concurrency 100
5:54PM INF ⏳Running quantitative tests with 100 goroutines
5:54PM DBG Using paranoia level: 2
5:54PM DBG Using default corpus cache directory: /home/esadc/.ftw
5:54PM DBG Preparing download of corpus file from https://downloads.wortschatz-leipzig.de/corpora/eng_news_2023_100K.tar.gz
5:54PM DBG filename eng_news_2023_100K-sentences.txt already exists
5:54PM DBG **> rule 932271 (PL2) with payload 50 =>  #0: uri_payload:a 1 million-ounce detour lake operation would put agnico eagles total gold production at ~4 million ounces which is knocking on the door of barrick gold.
5:54PM DBG **> rule 942390 (PL2) with payload 124 =>  #0: uri_payload:"Aap bas mujhe 200rs, 300rs de dena or 1000rs ke khane ke maje lena," Sati quoted the agent as saying in the post.
5:54PM DBG **> rule 932236 (PL2) with payload 128 =>  #0: uri_payload:Aaron Connolly spun a shot wide at the start of the second half and from there on in, it felt like a matter of time until the hosts netted a third which they duly did with 25 minutes left when Marcus Harness added some gloss to a fine display.
5:54PM DBG **> rule 932236 (PL2) with payload 176 =>  #0: uri_payload:Abdullahi said that the strike would affect the socio-economic activities of the country negatively, hence the need for his appeal for more time and realistic dialogue.
5:54PM DBG **> rule 932240 (PL2) with payload 197 =>  #0: uri_payload:Aberdeen secondary schools football: Goalkeeper O'Kennedy's heroics send Cults Academy first year into Green… #1: ARGS:uri_payload:Aberdeen secondary schools football: Goalkeeper O'Kennedy's heroics send Cults Academy first year into Green…
5:54PM DBG **> rule 932236 (PL2) with payload 205 =>  #0: uri_payload:A bid to give the public more time to consider borrowing $241 million to improve recreational facilities, fell flat at council’s regular Monday (July 24) meeting.
5:54PM DBG **> rule 942200 (PL2) with payload 63 =>  #0: uri_payload:A21 in both directions re-opened, traffic easing, earlier accident investigation work from B2160 Maidstone Road (Kipping's Cross Roundabout, Kipping's Cross) to A262 (Forstal Farm Roundabout, Lamberhurst). #1: ARGS:uri_payload:A21 in both directions re-opened, traffic easing, earlier accident investigation work from B2160 Maidstone Road (Kipping's Cross Roundabout, Kipping's Cross) to A262 (Forstal Farm Roundabout, Lamberhurst).
```

I tried downgrading to older versions of ftw versions but the issue exists there too.

OS: WSL Ubuntu 24.04
FTW Version: 2.3.0

Logs:

[trace logs.zip](https://github.com/user-attachments/files/28139739/trace.logs.zip)
